graylog2-server/src/main/java/org/graylog2/rest/resources/system/jobs/SystemJobResource.java

```diff
@@ -170,5 +171,35 @@ public Response trigger(@ApiParam(name = "JSON body", required = true)
         return Response.accepted().build();
     }
 
-    // TODO: DELETE: attempt to stop/cancel job
+    @DELETE
+    @Timed
+    @Path("/{jobId}")
+    @ApiOperation(value = "Cancel running job")
+    @Produces(MediaType.APPLICATION_JSON)
+    public SystemJobSummary cancel(@ApiParam(name = "jobId", required = true) @PathParam("jobId") @NotEmpty String jobId) {
+        SystemJob systemJob = systemJobManager.getRunningJobs().get(jobId);
+        if (systemJob == null) {
+            throw new NotFoundException("No system job with ID <" + jobId + "> found");
+        }
+
+        checkPermission(RestPermissions.SYSTEMJOBS_DELETE, systemJob.getClassName());
+
+        if (systemJob.isCancelable()) {
+            systemJob.requestCancel();
+        } else {
+            throw new ClientErrorException("System job with ID <" + jobId + "> cannot be cancelled", 422);
```

Comment:
Why respond with [HTTP status 422 (Unprocessable Entity)](https://httpstatuses.com/422) instead of 404 (Not Found) for example?